### PR TITLE
fix sign in code to generate a random signed short

### PR DESCRIPTION
**Pull request by pjfanning**
see https://github.com/apache/incubator-pekko/pull/401#discussion_r1231609526

### CODE REVIEW

**Review by He-Pin — APPROVED** [review]
lgtm

File: actor/src/main/scala/org/apache/pekko/io/dns/IdGenerator.scala [hunk 1]
```diff
@@ -30,7 +30,9 @@ private[pekko] trait IdGenerator {
  */
 @InternalApi
 private[pekko] object IdGenerator {
-  private val MaxUnsignedShort = 65535
+  // Random.nextInt(bound) generates a random int in the range 0 (inclusive) to bound (exclusive),
+  // so add 1 to Max Unsigned Short (65535)
+  private val SignedShortBound = 65536
```
Comment (nvollmar) [inline actor/src/main/scala/org/apache/pekko/io/dns/IdGenerator.scala:35]: Wouldn't that be `UnsignedShortBound`?

Comment (pjfanning) [reply actor/src/main/scala/org/apache/pekko/io/dns/IdGenerator.scala]: thanks - sorted that now

File: actor/src/main/scala/org/apache/pekko/io/dns/IdGenerator.scala [hunk 2]
```diff
@@ -57,5 +59,5 @@ private[pekko] object IdGenerator {
    * @return a random sequence of ids for production
    */
   def random(rand: java.util.Random): IdGenerator =
-    () => (rand.nextInt(MaxUnsignedShort) - Short.MinValue).toShort
+    () => (rand.nextInt(SignedShortBound) + Short.MinValue).toShort
```
[the remark this answers is not in the document]
Comment (pjfanning) [reply actor/src/main/scala/org/apache/pekko/io/dns/IdGenerator.scala]: 0 is valid. so `rand.nextInt(65536) will give us a response int between 0 and 65535 - ie 65536 possible answers. This is what I'm looking for.

65535 + Short.MIN_VALUE (-32768) is 32767 (which is Short.MAX_VALUE).

0 + Short.MIN_VALUE is -32768 (Short.MIN_VALUE).

**Pull request merged**